### PR TITLE
add puex store and vuex websockets

### PR DESCRIPTION
**Pull request by OmgImAlexis**
These are all libs apart from the `store.js`.

### CODE REVIEW

File: themes/dark/assets/js/lib/vuex.js [hunk 1]
```diff
@@ -0,0 +1,938 @@
+/**
```
Comment (sharkykh) [inline themes/dark/assets/js/lib/vuex.js]: Looks like this was committed by mistake? None of the other themes have this file, and it's not imported in `main.mako`.

File: themes-default/slim/static/js/store.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            });
+        }
+    },
+    plugins: []
+});
+
+Vue.use(VueNativeSock, 'ws://localhost:8081/x/ws/ui', {
+    store: window.store,
+    format: 'json'
+});
```
Comment (sharkykh) [inline themes-default/slim/static/js/store.js]: Do you know if doing this makes it try to connect to the websocket?
Do you need to add `connectManually: true,` to disable that?

File: themes-default/slim/static/js/store.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                    release: null,
+                    aliases: null,
+                    subtitlesEnabled: null,
+                    anime: null
+                }
+            }
+        }
+    },
+    mutations: {
+        [LOGIN_PENDING]() {},
```
Comment (sharkykh) [inline themes-default/slim/static/js/store.js]: What sorcery is this? :scream: 
Can you explain why it's in brackets?

Comment (OmgImAlexis) [reply themes-default/slim/static/js/store.js]: const x = 'a';[x]

Comment (OmgImAlexis) [reply themes-default/slim/static/js/store.js]: const x = 'a';
const y = {};[a] =

**Comment by OmgImAlexis** [comment]
@p0psicles any ideas if we pass anything but JSON?

Just noticed this section with `{ format: json }` enabled.
https://github.com/nathantsoi/vue-native-websocket#with-format-json-enabled

**Comment by sharkykh** [comment]
@OmgImAlexis 
We **currently** don't pass anything but JSON through the websocket.
And frankly I don't think we will ever pass anything else.

**Comment by sharkykh** [comment]
Just a small note: It currently opens two WebSocket connections.
Can we move the notification handling to this file right now?

**Comment by OmgImAlexis** [comment]
@sharkykh yes that was my plan after merging this.

**Pull request merged**